src/Symfony/Bundle/TwigBundle/DependencyInjection/Compiler/RuntimeLoaderPass.php

```diff
@@ -33,7 +33,18 @@ public function process(ContainerBuilder $container): void
             $def = $container->getDefinition($id);
             $mapping[$def->getClass()] = new Reference($id);
         }
+        $classesWithAttribute = [];
+        foreach ($container->findTaggedServiceIds('twig.attribute_extension', true) as $id => $attributes) {
+            $def = $container->getDefinition($id);
+            $mapping[$def->getClass()] = new Reference($id);
```

Comment:
This gives precedence to attributes. This is less flexible because one cannot remove an attribute from a class. ;)

Reply:
`twig.attribute_extension` implies `twig.runtime` and cannot work without it. The conflict would be if there is 2 services with a different id, but the same class name.  One is a `twig.runtime`, the other a `twig.attribute_extension`.

Do you think the priority should be reversed like this?
```suggestion
            $mapping[$def->getClass()] ??= new Reference($id);
```